examples/advanced/spotlight/main.py

```diff
@@ -0,0 +1,40 @@
+import blenderproc as bproc
+import numpy as np
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument('scene', nargs='?', default="examples/resources/scene.obj", help="Path to the scene.obj file")
+parser.add_argument('output_dir', nargs='?', default="examples/advanced/spotlight/output",
+                    help="Path to where the final files, will be saved")
+args = parser.parse_args()
+
+bproc.init()
+
+# load the objects into the scene
+objs = bproc.loader.load_obj(args.scene)
+
+# Find point of interest, all cam poses should look towards it
+poi = bproc.object.compute_poi(objs)
+# Sample five camera poses
+for i in range(3):
+    # Sample random camera location above objects
+    location = np.random.uniform([-10, -10, 8], [10, 10, 12])
+    # Compute rotation based on vector going from location towards poi
+    rotation_matrix = bproc.camera.rotation_from_forward_vec(poi - location,
+                                                             inplane_rot=np.random.uniform(-0.7854, 0.7854))
+    # Add homog cam pose based on location a rotation
+    cam2world_matrix = bproc.math.build_transformation_mat(location, rotation_matrix)
+    bproc.camera.add_camera_pose(cam2world_matrix)
+
+# add a spotlight which intersect with the sampled camera poses
+spot_light = bproc.lighting.add_intersecting_spot_lights_to_camera_poses(clip_start=7, clip_end=15)
```

Comment:
Why is clip_start 7? Isn't it possible that everything is much closer and the light is sampled completely away from all objects?

Reply:
These values worked nicely for the current scene. :D That depends on the camera position and the range of the objects. So far I had no idea how to abstract these two numbers to make it more general.

Reply:
Maybe one could sample the look at point via sending rays from the camera similar to how we check visible objects in camera sampling. But this is out of the scope of this PR